R/utils.R

```diff
@@ -243,16 +243,24 @@ ReadFTPC <- function(conn) {
 
   #Events (e.g. The date the plot was sampled, QA/QC records)
 
-  #Short
-  tbl_Events_short <- dplyr::tbl(conn, "tbl_Events") %>%
-    dplyr::select(Event_ID, Plot_ID, Start_Date, QA_Plot)
+  tbl_Events <- dplyr::tbl(conn, "tbl_Events") %>%
+    dplyr::select(Event_ID, Plot_ID, Start_Date, QA_Plot, Images, Max_Veg_Ht = Max_veg_ht,
+                  Entered_Date = Entered_date, Updated_Date = Updated_date, Verified, Verified_By = Verified_by, Verified_Date = Verified_date,
+                  Certified, Certified_By = Certified_by, Certified_Date = Certified_date, Completion_Time = Completion_time,
+                  Event_Notes, QA_notes) %>%
+    # add "Year" (year sampled) and "Cycle" (sample cycle)
+    dplyr::mutate(Year = YEAR(Start_Date)) %>%
+    #supposedly SQL does not have translation for case_when so must use "ifelse" ??
+    dplyr::mutate(Cycle = ifelse(Year <= 2014, 1, NA)) %>%
+    dplyr::mutate(Cycle = ifelse(is.na(Cycle) & Year >= 2015 & Year <= 2020, 2, Cycle)) %>%
+    dplyr::mutate(Cycle = ifelse(is.na(Cycle) & Year >= 2021, 3, Cycle))
```

Comment:
I'd consider defaulting cycle to NA at some point so that visits don't get assigned cycle 3 forever into the future